src/entity.ts

```diff
@@ -130,7 +150,46 @@ export namespace entity {
        * @name Int#value
        * @type {string}
        */
-      this.value = value.toString();
+      if (typeCastOptions) {
+        this.typeCastFunction = typeCastOptions.integerTypeCastFunction;
+        if (typeof typeCastOptions.integerTypeCastFunction !== 'function') {
+          throw new Error(
+            `integerTypeCastFunction is not a function or is not provided.`
+          );
+        }
+
+        this.typeCastProperties = typeCastOptions.properties
+          ? arrify(typeCastOptions.properties)
+          : undefined;
+      }
+    }
+    // tslint:disable-next-line no-any
+    valueOf(): any {
+      let shouldCustomCast = this.typeCastFunction ? true : false;
+      if (
+        this.typeCastProperties &&
+        !this.typeCastProperties.includes(this._entityPropertyName!)
+      ) {
+        shouldCustomCast = false;
+      }
+
+      if (shouldCustomCast) {
+        try {
+          return this.typeCastFunction!(this.value);
+        } catch (error) {
+          error.message = `integerTypeCastFunction threw an error:\n\n - ${error.message}`;
```

Comment:
Could you do two spaces in front of the hyphen? I think that’s the standard newline+indent formatting we use in other places.

Reply:
Done